lib/subordinateio.c

```diff
@@ -307,39 +307,23 @@ static bool have_range(struct commonio_db *db,
 	return false;
 }
 
-static bool append_range(struct subordinate_range ***ranges, const struct subordinate_range *new, int n)
+static bool append_range(struct subordinate_range **ranges, const struct subordinate_range *in, int n)
 {
-	struct subordinate_range *tmp;
 	if (!*ranges) {
-		*ranges = malloc(sizeof(struct subordinate_range **));
+		*ranges = malloc(sizeof(struct subordinate_range));
 		if (!*ranges)
 			return false;
 	} else {
-		struct subordinate_range **new;
-		new = realloc(*ranges, (n + 1) * (sizeof(struct subordinate_range **)));
+		struct subordinate_range *new;
+		new = realloc(*ranges, (n + 1) * (sizeof(struct subordinate_range)));
 		if (!new)
 			return false;
 		*ranges = new;
 	}
-	(*ranges)[n] = NULL;
-	tmp = subordinate_dup(new);
-	if (!tmp)
-		return false;
-	(*ranges)[n] = tmp;
+	(*ranges)[n] = *in;
 	return true;
 }
 
-void free_subordinate_ranges(struct subordinate_range **ranges, int count)
-{
-	int i;
-
-	if (!ranges)
-		return;
-	for (i = 0; i < count; i++)
-		subordinate_free(ranges[i]);
```

Comment:
`subordinate_free()` seems to be unused anymore.

Reply:
It's still used as a member of the commonio_ops struct.

Reply:
This member wasn't used in that version of PR. But doesn't matter already.